lib/std/simd.zig

```diff
@@ -278,40 +278,113 @@ test "vector shifting" {
     try std.testing.expectEqual([4]u32{ 40, 30, 20, 10 }, reverseOrder(base));
 }
 
+/// Return the index of the first true bool in `vec` or null if there
+/// is none.
 pub fn firstTrue(vec: anytype) ?VectorIndex(@TypeOf(vec)) {
     const len = vectorLength(@TypeOf(vec));
-    const IndexInt = VectorIndex(@TypeOf(vec));
 
-    if (!@reduce(.Or, vec)) {
-        return null;
+    // FIXME: bit casting between bool vectors and ints differs in CBE
```

Comment:
I would prefer a github issue link to a TODO comment here.

The comment mentions an issue with the C backend, but the if condition is testing for the LLVM backend. Did you mean to test for the C backend instead? 

https://github.com/ziglang/zig/blob/7ab306d1f7c3f028ff2d90cb32343c2f2444a86f/lib/std/builtin.zig#L665-L708

Reply:
I test for the LLVM backend to ensure that this is only used for backends that implement `@bitCast` between bool vectors and ints the same way as LLVM does, otherwise it relies on the "guaranteed" builtins. As far as I can tell, no other backend is far enough to have implemented this yet. I'll update the comment accordingly.